src/uWS.h

```diff
@@ -14,8 +33,51 @@
 #include <Windows.h>
 #endif
 
+#include <uv.h>
+
 namespace uWS {
 
+class MutexImpl {
+private:
+#if USE_WIN32_THREADS
+  CRITICAL_SECTION m_mutex;
+#else
+  std::mutex m_mutex;
+#endif
+  
+  MutexImpl(const MutexImpl& other) = delete;
+  MutexImpl& operator=(const MutexImpl& other) = delete;
+
+public:
+  MutexImpl() {
+#if USE_WIN32_THREADS
+    InitializeCriticalSection(&m_mutex);
+#endif
```

Comment:
Like I said, from Googling the problem, StackOverflow says that you should use .NET threading when in C++/CLI. This is using the Win32 threading functions.